# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/WebCodecsUtils.h
# Commit: 1cf50ce36a73
# Full Hash: 1cf50ce36a7331c58822b4caecb2a67103eb6da5
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-29 15:31:45
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Guess pixel format from VideoData r=aosmond
#   
#   Depends on D170635
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171166
# ==============================================================================

diff -r d7f55df2c2e4 -r 1cf50ce36a73 dom/media/webcodecs/WebCodecsUtils.h
--- a/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 04:15:03 2023 +0000
+++ b/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 04:15:03 2023 +0000
@@ -21,6 +21,7 @@
 namespace gfx {
 enum class ColorRange : uint8_t;
 enum class ColorSpace2 : uint8_t;
+enum class SurfaceFormat : int8_t;
 enum class TransferFunction : uint8_t;
 enum class YUVColorSpace : uint8_t;
 }  // namespace gfx
@@ -79,6 +80,20 @@
 
 Maybe<VideoColorPrimaries> ToPrimaries(const gfx::ColorSpace2& aColorSpace);
 
+/*
+ * The following are utilities to convert from gfx's formats to
+ * VideoPixelFormats.
+ */
+
+enum class ImageBitmapFormat : uint8_t;
+enum class VideoPixelFormat : uint8_t;
+
+Maybe<VideoPixelFormat> SurfaceFormatToVideoPixelFormat(
+    gfx::SurfaceFormat aFormat);
+
+Maybe<VideoPixelFormat> ImageBitmapFormatToVideoPixelFormat(
+    ImageBitmapFormat aFormat);
+
 }  // namespace dom
 }  // namespace mozilla
 
